### ISSUE TITLE
Error on google.protobuf.Empty

### ISSUE THREAD

**Issue by drasko** [tomas-abrahamsson/gpb#170]
I am getting error:
```
in service ThingsService, rpc CanAccessByID, return: undefined reference google.protobuf.Empty
```
on trying to compile proto files:
```
./gpb/bin/protoc-erl -I ./gpb/ ../../*.proto
```
which are basically located [here](https://github.com/Mainflux/mainflux).

Main problem is this file: https://github.com/mainflux/mainflux/blob/master/internal.proto which contains line:
```
import "google/protobuf/empty.proto";
```
and then line:
```
rpc CanAccessByID(AccessByIDReq) returns (google.protobuf.Empty) {}
```
which uses it.

It looks like `gpb` does not know how to handle `google.protobuf.Empty`.

**Comment by tomas-abrahamsson** [comment]
Hi, try adding `-pkgs` to the options to `protoc-erl`:
```
./gpb/bin/protoc-erl -pkgs -I ./gpb/ ../../*.proto
```

The default for gpb is to ignore packages, for historical reasons, but especially once you include things from another package, such as `google.protobuf.Empty` you need to tell gpb to consider package namespaces.

**Comment by drasko** [comment]
@tomas-abrahamsson thanks a lot for the quick help - that worked.

Maybe this should be noted in the doc, because I did not spot it over there?

I am closing the issue.

**Comment by drasko** [comment]
One more related question - I have opened the issue for `grpc_client` as well: https://github.com/Bluehouse-Technology/grpc_client/issues/13

From what I see here: https://github.com/Bluehouse-Technology/grpc_client/blob/master/src/grpc_client.erl#L122 and here: https://github.com/Bluehouse-Technology/grpc_lib/blob/master/src/grpc_lib_compile.erl#L43 and options can be passed to `gpb` in a form of map.

How to pass this `-pkgs` option, i.e. how to tell to `grpc_client` to use `gpb` with `-pkgs` enabled?

**Comment by drasko** [comment]
OK, found it.

Something like this:
```
grpc_client:compile("../../../internal.proto", [{use_packages, true}]).
```
did the work.

I took a look at the code here: https://github.com/tomas-abrahamsson/gpb/blob/0a3358238327ffbe84098223ed28ec91e49eff64/src/gpb_compile.erl#L1881 and  here: https://github.com/tomas-abrahamsson/gpb/blob/0a3358238327ffbe84098223ed28ec91e49eff64/src/gpb_compile.erl#L143 to understand what's going on.

**Comment by tomas-abrahamsson** [comment]
> Maybe this should be noted in the doc, because I did not spot it over there?

This has been a recurring question, but since it can actually be fairly well detected, I added to gpb a while back that it automatically hint about the option (in commit e107a1f587). But unfortunately, I forgot about the case when it is referred to only from rpc args or return values, which was the case you had here. So on my local master, I've now added hinting for that case too. It will look like this once it lands in the next release:
```
$ protoc_erl ... internal.proto
hint: use the option use_packages (-pkgs) to use messages or enums in other packages
in service ThingsService, rpc CanAccessByID, return: undefined reference google.protobuf.Empty
```

> How to pass this `-pkgs` option
> OK, found it.

Yes exactly, `use_packages` or equivalently `{use_packages, true}` is the corresponding erlang-level option. Looks like you've got it right.
